staging/src/k8s.io/endpointslice/reconciler.go

```diff
@@ -253,29 +253,22 @@ func (r *Reconciler) reconcileByAddressType(logger klog.Logger, service *corev1.
 	// Topology hints are assigned per address type. This means it is
 	// theoretically possible for endpoints of one address type to be assigned
 	// hints while another endpoints of another address type are not.
-	si := &topologycache.SliceInfo{
+	si := &topologyheuristics.SliceInfo{
 		ServiceKey:  fmt.Sprintf("%s/%s", service.Namespace, service.Name),
 		AddressType: addressType,
 		ToCreate:    slicesToCreate,
 		ToUpdate:    slicesToUpdate,
 		Unchanged:   unchangedSlices(existingSlices, slicesToUpdate, slicesToDelete),
 	}
 
-	if r.topologyCache != nil && hintsEnabled(service.Annotations) {
-		slicesToCreate, slicesToUpdate, events = r.topologyCache.AddHints(logger, si)
+	_, hintsEnabled := topologyheuristics.HeuristicNameFromAnnotations(service.Annotations)
```

Comment:
the boolean return is not really needed, it seems more a commodity but is overloading the method, we can just add 3 lines more here and get rid of it

Reply:
you can just keep the existing function in `staging/src/k8s.io/endpointslice/utils.go`